src/plugins/encryption/PasswordEncryptionHandler.h

```diff
@@ -0,0 +1,28 @@
+//
+// Created by moorts on 25.10.21.
+//
+
+#ifndef SRC_PASSWORDENCRYPTIONHANDLER_H
+#define SRC_PASSWORDENCRYPTIONHANDLER_H
+
+#include "KeyDerivation.h"
+
+template <class C>
+concept CIPHER = requires(C) {
+    { C::encrypt(NULL, NULL) } -> std::same_as<std::string>;
+    { C::decrypt(NULL, NULL) } -> std::same_as<std::string>;
+};
+
+template <CIPHER C>
+class PasswordEncryptionHandler {
```

Comment:
What's the difference between this class and EncryptonInterface? It seems to be doing the same only that this class adds the concept. One of those two classes should be enough then, right?